aux/javapackages-tools.spec

```diff
@@ -19,133 +34,141 @@ Source0:        javapackages-%{version}.tar.gz
 resolution using XMvn resolver.
 
-%package -n ivy-local
+%package -n %{?scl_prefix}ivy-local
 Summary:        Local mode for Apache Ivy
 Requires:       %{name} = %{version}-%{release}
-Requires:       javapackages-local = %{version}-%{release}
-Requires:       apache-ivy >= 2.3.0-8
-Requires:       xmvn-connector-ivy >= 2
+Requires:       %{?scl_prefix}javapackages-local = %{version}-%{release}
+Requires:       %{?scl_prefix}apache-ivy >= 2.3.0-8
+Requires:       %{?scl_prefix}xmvn-connector-ivy >= 2
 
-%description -n ivy-local
+%description -n %{?scl_prefix}ivy-local
 This package implements local mode for Apache Ivy, which allows
 artifact resolution using XMvn resolver.
 
-%package -n python3-javapackages
+%package -n %{?scl_prefix}%{python_prefix}-javapackages
 Summary:        Module for handling various files for Java packaging
-Requires:       python3-lxml
-Requires:       python3-six
-Obsoletes:      python-javapackages < %{version}-%{release}
+Requires:       %{python_prefix}-lxml
+Requires:       %{python_prefix}-six
+Obsoletes:      %{?scl_prefix}python-javapackages < %{version}-%{release}
 
-%description -n python3-javapackages
+%description -n %{?scl_prefix}%{python_prefix}-javapackages
 Module for handling, querying and manipulating of various files for Java
 packaging in Linux distributions
 
-%package -n javapackages-local
+%package -n %{?scl_prefix}javapackages-local
 Summary:        Non-essential macros and scripts for Java packaging support
 Requires:       %{name} = %{version}-%{release}
-Requires:       xmvn-install >= 2
-Requires:       xmvn-subst >= 2
-Requires:       xmvn-resolve >= 2
+%if ! 0%{?bootstrap}
+Requires:       %{?scl_prefix}xmvn-install >= 2
+Requires:       %{?scl_prefix}xmvn-subst >= 2
+Requires:       %{?scl_prefix}xmvn-resolve >= 2
+%endif
 # Java build systems don't have hard requirement on java-devel, so it should be there
 Requires:       java-devel
 
-%description -n javapackages-local
+%description -n %{?scl_prefix}javapackages-local
 This package provides non-essential macros and scripts to support Java packaging.
 
 %prep
-%setup -q
+%setup -q -n %{pkg_name}-%{version}
 
 sed -i '/fedora-review/d' install
 
+# Add SCL namespace to generated provides
+%{?scl: sed -i '/<groupId>/{h;s|<.*|<namespace>%{scl}</namespace>|;p;g}' etc/javapackages-metadata.xml}
+
 %build
-%configure --pyinterpreter=%{__python3}
+%{?scl:scl enable %{scl} - << "EOF"}
+set -e -x
+%configure --pyinterpreter=%{python_interpreter}
 ./build
+%{?scl:EOF}
 
 %install
 ./install
-sed -i 's|mvn_build.py|& --xmvn-javadoc|' %{buildroot}/usr/lib/rpm/macros.d/macros.fjava
+
+sed -i 's|mvn_build.py|& --xmvn-javadoc|' $(find %{buildroot} -name macros.fjava)
 sed -e 's/.[17]$/&.gz/' -e 's/.py$/&*/' -i files-*
 
-pushd python
-  %{__python3} setup.py install -O1 --skip-build --root %{buildroot}
-popd
+%{?scl: sed -i 's:${rpmconfigdir}/macros.d:%{_root_sysconfdir}/rpm:' install}
```

Comment:
When in SCL we must install RPM macros to root /etc directory, otherwise rpm wouldn't be able to load it. All paths known to the install script have /opt/... prefix, so it can't be simply done by patching it.